src/libpmemset/file.c

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * file.c -- implementation of common file API
+ */
+
+#include <libpmem2.h>
+
+#include "alloc.h"
+#include "file.h"
+#include "pmemset_utils.h"
+
+struct pmemset_file {
+	bool close;
```

Comment:
Is "close" good choice for the indicator name?  If I understand purpose of it, something like "is_open" would be better. Up to you.

Reply:
When the pmemset_file structure is created, the file is always open. It is either opened manually to create pmem2_source from fd/handle and pmemset_file from this source or internally when creating pmemset_file from the provided path. "close" member indicates whether the file should be internally closed when deleting the pmemset_file struct, which is not the case when you create pmemset_file from pmem2_source.